src/Clapi/TextSerialisation.hs

```diff
@@ -24,10 +30,49 @@ tpBuilder ((Time s f), vs) = tb <> mconcat (map (\v -> (fromChar ' ') <> (cvBuil
   where
     tb = (fss s) <> (fromChar ':') <> (fss f)
 
-encode :: Map.Map Time [ClapiValue] -> Builder
+type TimeSeries = Map.Map Time [ClapiValue]
+
+encode :: TimeSeries -> Builder
 encode ts = header <> bodyBuilder
   where
     -- It is invalid for a time series to be empty, so this use of head is
     -- kinda fine, but the errors will suck:
     header = fromString $ map typeTag $ head $ Map.elems ts
     bodyBuilder = mconcat $ map (\tp -> fromChar '\n' <> tpBuilder tp) (Map.toList ts)
+
+quotedString :: Parser Text
+-- FIXME: escaped quotes unsupported
```

Comment:
There _must_ be a pre-canned parser somewhere that someone has made for grabbing a quoted string and handling escapes...

Reply:
I can't find one, but I stink at finding haskell things. It isn't particularly tricky since attoparsec has a stateful take thing which is what you need.